src/libcrun/criu.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * crun - OCI runtime written in C
+ *
+ * Copyright (C) 2020 Adrian Reber <areber@redhat.com>
+ * crun is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * crun is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with crun.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#define _GNU_SOURCE
+
+#include <config.h>
+
+#ifdef HAVE_CRIU
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <criu/criu.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "container.h"
+#include "error.h"
+#include "status.h"
+#include "utils.h"
+
+#define CRIU_LOG_FILE "dump.log"
+
+int
+libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status,
+					 libcrun_container_t *container,
+					 libcrun_checkpoint_restore_t *
+					 cr_options, libcrun_error_t *err)
+{
+  runtime_spec_schema_config_schema *def = container->container_def;
+  cleanup_free char *path = NULL;
+  cleanup_close int image_fd;
+  cleanup_close int work_fd;
```

Comment:
please initialize these fds to -1, otherwise if we return earlier from the function without initializing them, the auto cleanup function will call `close` on the unitialized value found on the stack.

Reply:
done